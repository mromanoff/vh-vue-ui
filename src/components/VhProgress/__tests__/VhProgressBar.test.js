import { shallowMount } from '@vue/test-utils';
import VhProgressBar from '../VhProgressBar.vue';

describe('VhProgressBar.vue', () => {
  let wrapper;

  it('renders default VhProgressBar', () => {
    wrapper = shallowMount(VhProgressBar);
    expect(wrapper.props().color).toBe('orange');
    expect(wrapper.props().size).toBe('medium');
    expect(wrapper.props().value).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhProgressBar blue large', () => {
    wrapper = shallowMount(VhProgressBar, {
      propsData: {
        color: 'blue',
        size: 'large',
      },
    });
    expect(wrapper.props().color).toBe('blue');
    expect(wrapper.props().size).toBe('large');
    expect(wrapper.props().value).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhProgressBar green small', () => {
    wrapper = shallowMount(VhProgressBar, {
      propsData: {
        color: 'green',
        size: 'small',
        value: 10,
      },
    });
    expect(wrapper.props().color).toBe('green');
    expect(wrapper.props().size).toBe('small');
    expect(wrapper.props().value).toBe(10);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhProgressBar blue small', () => {
    wrapper = shallowMount(VhProgressBar, {
      propsData: {
        color: 'blue',
        size: 'medium',
        value: 1,
      },
    });
    expect(wrapper.props().color).toBe('blue');
    expect(wrapper.props().size).toBe('medium');
    expect(wrapper.props().value).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
