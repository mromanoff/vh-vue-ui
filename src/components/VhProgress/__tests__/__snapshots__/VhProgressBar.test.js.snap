// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VhProgressBar.vue renders default VhProgressBar 1`] = `
<div role="presentation" class="Progress Progress--medium">
  <div class="Progress-track">
    <div class="Progress-bar Progress-bar--orange" style="width: 0%;"></div>
  </div>
</div>
`;

exports[`VhProgressBar.vue renders modified VhProgressBar blue large 1`] = `
<div role="presentation" class="Progress Progress--large">
  <div class="Progress-track">
    <div class="Progress-bar Progress-bar--blue" style="width: 0%;"></div>
  </div>
</div>
`;

exports[`VhProgressBar.vue renders modified VhProgressBar blue small 1`] = `
<div role="presentation" class="Progress Progress--medium">
  <div class="Progress-track">
    <div class="Progress-bar Progress-bar--blue" style="width: 1%;"></div>
  </div>
</div>
`;

exports[`VhProgressBar.vue renders modified VhProgressBar green small 1`] = `
<div role="presentation" class="Progress Progress--small">
  <div class="Progress-track">
    <div class="Progress-bar Progress-bar--green" style="width: 10%;"></div>
  </div>
</div>
`;
